src/Commands.hs

```diff
@@ -850,3 +853,27 @@ saveDocs :: Context -> [(SymPath, Binder)] -> IO (Context, Either a XObj)
 saveDocs ctx pathsAndEnvBinders = do
      liftIO (saveDocsForEnvs (contextProj ctx) pathsAndEnvBinders)
      return (ctx, dynamicNil)
+
+commandInlineC :: CommandCallback
+commandInlineC ctx (XObj (Sym p _) _ _ : args) =
+  case mapM (checkArg ctx) args of
+    Just (err:_) -> return err
+    Nothing -> do
+      let src = concatMap stringify args
+          registration = XObj (InlinedC ("// " ++ show p ++ "\n" ++ src)) Nothing Nothing
+          globalEnv = contextGlobalEnv ctx
+          env' = envInsertAt globalEnv p (Binder emptyMeta registration)
+      return (ctx { contextGlobalEnv = env' }, dynamicNil)
+  where
+    --checkArg _ (XObj (Sym _ _) _ _) = Nothing
+    checkArg _ (XObj (Num _ _) _ _) = Nothing
+    checkArg _ (XObj (Str _) _ _) = Nothing
+    checkArg ctx x = Just (evalError ctx ("Invalid argument to inline-c (expected Num, Str, Sym, InlinedC) : " ++ pretty x) (info x))
+
+    stringify :: XObj -> String
+    --stringify (XObj (Sym p _) _ _) = show p
+    stringify x@(XObj (Num _ _) _ _) = pretty x
+    stringify (XObj (Str s) _ _) = unescapeString s
+
+commandInlineC ctx _ =
+  return (evalError ctx "inline-c expects a Symbol then some Strings and/or Numbers." (Just dummyInfo))
```

Comment:
I would consider matching on the arguments here and changing the error message to:

```
"The first argument to `inline-c` must be a symbol, got `"++ <matchvar> ++ "`"
```